python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -164,15 +169,24 @@ def __really_writeIndex(self):
 
         raw_self = stripProxy(self)
 
+        import os
+
         all_caches = {}
-        for i in range(len(raw_self)):
-            this_cache = raw_self._registry.getIndexCache( stripProxy(raw_self.__getitem__(i)) )
-            all_caches[i] = this_cache
-            disk_location = raw_self.__get_dataFile(i)
-            import os
-            all_caches[i]['modified'] = os.stat(disk_location).st_ctime
+        for sj_id in range(len(raw_self)):
```

Comment:
If you're sure that the enumerate function only calls ```__len__``` and only explicitly calls ```__getitem__``` when required then I'm fine with that but I'd need to check as the ```__getitem__`` loads the subjob and hence if it's called rather than ```__len__``` it will break what the code is trying to do.